Set probabilistic sampling as default subsampling behavior

huddlej opened the pull request:
This PR makes probabilistic sampling the default behavior for `augur filter` when the user requests a maximum number of subsampling sequences.

Major changes include:

  - Adds a `--no-probabilistic-sampling` flag that mirrors the existing `--probabilistic-sampling` flag. This flag forces augur to try to accomodate no more than the request number of sequences and throw an error if there are more subsampling groups than available sequences.
 - Makes `--probabilistic-sampling` the default behavior when users request a maximum number of sequences to subsample. This change to the filter API allows users to define subsampling groups whose number exceeds the maximum sequences requested without throwing an error. As described in the conversation of issue #654, this new behavior is more likely the expected behavior by most users.

Minor changes include:

 - Removes the Zika Snakemake functional tests. These tests could have been removed back when we originally migrated them to the Cram functional tests (`zika.t`). Since this PR revealed the annoying CI errors that can result from maintaining redundant functional tests, it is time to remove these older tests.
  - Explicitly disables probabilistic sampling in functional tests random output impedes testing for correctness.

Fixes #654

rneher commented on augur/filter.py:
```diff
@@ -100,7 +100,9 @@ def register_arguments(parser):
-    parser.add_argument('--probabilistic-sampling', action='store_true', help="Sample probabilitically from groups -- useful when there are more groups than requested sequences")
+    probabilistic_sampling_group = parser.add_mutually_exclusive_group()
+    probabilistic_sampling_group.add_argument('--probabilistic-sampling', action='store_true', default=True, help="Enable probabilistic sampling during subsampling. This is useful when there are more groups than requested sequences. This option only applies when `--subsample-max-sequences` is provided.")
+    probabilistic_sampling_group.add_argument('--no-probabilistic-sampling', action='store_true', help="Disable probabilistic sampling during subsampling, requiring that there must be no more subsampling groups than the requested maximum number of sequences. This option only applies when `--subsample-max-sequences` is provided.")
```
one could use 

`probabilistic_sampling_group.add_argument('--no-probabilistic-sampling', action='store_false', dest='probabilistic_sampling', help="...")`

which should straighten out the logic below a bit.

rneher reviewed (commented):
thanks John. This looks good to me. I only have one optional suggestion for the argument handling.

huddlej commented:
Thank you, @rneher! I've updated the parser to use the `store_false` approach you recommended and squashed all of the commits into a single cleaner one. When CI tests finish, I'll merge this and tomorrow I will prepare a release (since there's a little too much going on today).
